### ISSUE TITLE
Tutorials

### ISSUE THREAD

**Issue by omoindrot** [cs230-stanford/cs230-code-examples#2]
*Content ideas for the tutorials explaining the posts*

## Content ideas for tutorials

Code in https://github.com/cs230-stanford/cs230-stanford.github.io

- structure of the project (files' roles, experiment pipeline)
- how to run the toy examples
- explain how to use logger
- explain where to define the model or change it
- explain how to change hyperparameters
- how to feed data...

- use github release to have multiple version of the code?

- Explain the general idea of training multiple models, trying different structures...
  - make sure that experiments are reproducible
    - for instance, if model.py has incompatible changes (ex: adds batch norm), previous params.json cannot be run again
    - have to update old params.json to match the new change (ex: put `params.use_bn` argument, and add it to all old `params.json`)
  - give good names to the dirs in `experiments`
  - visualize on tensorboard
  - don't spend too much time watching training progress: launch hyperparam search, let it run and get back later (make sure there is no bug first)

- explain how to properly split train / dev / test
  - hardcode the split in three folders

## Organization ideas

- add a number to each post? ex: "3. Creating input pipelines..."
  - would be easier to understand the structure
  - in each post, at the beginning put the full list
